### PR TITLE
Ticket 4112: NIMROD jaws manager

### PR DESCRIPTION
**Pull request by DominicOram**
See https://github.com/ISISComputingGroup/IBEX/issues/4112

To test:
* Run the system test in https://github.com/ISISComputingGroup/EPICS-IOC_Test_Framework/pull/195 against master. You will need to consider the following to make them work:
     - Polaris' settings will need to be updated as in this ticket
     - The PV for GEM's sample jaws has changed (which is dealt with in an upgrade script). You should still be able to run the tests by just changing the PV
* Run the same tests against this branch
* Confirm they pass

Documentation has been expanded for the generic case at https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Jaws-Managers and for individual beamlines linked from that ticket.

See https://github.com/ISISComputingGroup/EPICS-upgrade/commit/e18878be65a36a6b4ac84131f3419399978797cc for changes to GEM PV. 

Given the change in settings would only effect a single instrument it seemed overkill to add to the upgrade script so added note at https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev#update-information

- [ ] If you have updated the `settings` directory, have you added a corresponding step to the upgrade script to update affected instruments?

### CODE REVIEW

File: jawsManagerApp/Db/jaws_manager_single.template [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  field(INPC, "$(MOD_TO_SAMPLE)")
+  field(INPD, "$(MOD_OFFSET=0)")  
+  field(INPE, "$(MOD_TO_JAW)") 
+  
+  field(CLCK, "(A+((B-A)*((C-E)/(C-D))))")
+  
+  field(OUTK, "$(P)JAWMAN:$(JAW):VGAP:SP PP")
+}
+
+record(ao, "$(P)JAWMAN:$(JAW):HGAP:SP") {
```
Comment (Alistair-McGann-Tessella) [inline jawsManagerApp/Db/jaws_manager_single.template]: Do these records need to be archived? The delayed jaw values are archived

File: gemBeamlineJawsApp/Db/gem_beamline_jawset.substitutions [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	{DIR=V}
+}
+
+file "$(TOP)/jawsManagerApp/Db/jaws_manager_single.template" {
+    {JAW="JAWS1", MOD_TO_JAW="\$(MOD_TO_JAWS1)"}
+    {JAW="JAWS2", MOD_TO_JAW="\$(MOD_TO_JAWS2)"}
+    {JAW="JAWS3", MOD_TO_JAW="\$(MOD_TO_JAWS3)"}
+    {JAW="JAWS4", MOD_TO_JAW="\$(MOD_TO_JAWS4)"}
+	{JAW="JAWS5", MOD_TO_JAW="\$(MOD_TO_JAWS5)"}
+}
```
Comment (Alistair-McGann-Tessella) [inline gemBeamlineJawsApp/Db/gem_beamline_jawset.substitutions]: This needs newline at the end of the file

File: nimrodJawsApp/Db/nimrodjawset.substitutions [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+file "$(TOP)/jawsManagerApp/Db/jaws_manager_single.template" {
+    {JAW="JAWS1", MOD_TO_JAW="\$(MOD_TO_JAWS1)"}
+    {JAW="JAWS2", MOD_TO_JAW="\$(MOD_TO_JAWS2)"}
+    {JAW="JAWS3", MOD_TO_JAW="\$(MOD_TO_JAWS3)"}
+    {JAW="JAWS4", MOD_TO_JAW="\$(MOD_TO_JAWS4)"}
+	{JAW="JAWS5", MOD_TO_JAW="\$(MOD_TO_JAWS5)"}
+    {JAW="JAWS6", MOD_TO_JAW="\$(MOD_TO_JAWS6)"}
+}
```
Comment (Alistair-McGann-Tessella) [inline nimrodJawsApp/Db/nimrodjawset.substitutions]: Needs a newline

File: polarisJawsApp/Db/polarisjawset.substitutions [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    {JAW="JAWS1", MOD_TO_JAW="\$(MOD_TO_JAWS1)"}
+    {JAW="JAWS2", MOD_TO_JAW="\$(MOD_TO_JAWS2)"}
+    {JAW="JAWS3", MOD_TO_JAW="\$(MOD_TO_JAWS3)"}
+    {JAW="JAWS4", MOD_TO_JAW="\$(MOD_TO_JAWS4)"}
+    {JAW="JAWS5", UNDERLYING_JAW="\$(P)POLJAWSET:JAWS5:", MOD_TO_JAW="\$(MOD_TO_JAWS5)"}
+}
+
+file "$(TOP)/polarisJawsApp/Db/delayed_jaws.template" {
+	{TOP_LEVEL_JAW=POLJAWSET:JAWS5:, UNDERLYING_JAW=MOT:JAWS5:}
+}
```
Comment (Alistair-McGann-Tessella) [inline polarisJawsApp/Db/polarisjawset.substitutions]: Needs a newline

File: polarisJawsApp/Db/delayed_jaws.template [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+#
+#  Records that provide a wrapper around an alias that will store values and only send them on to the jawset when a set PV is issued
+#  These values can only be sent to the underlying jaw when in manager mode
+#
+#  Macros:
+#    P - base name for ioc
+#    TOP_LEVEL_JAW - the pv for the top level jaw
+#    UNDERLYING_JAW - the pv for the underlying jaw
+
+record(ao, "$(P)$(TOP_LEVEL_JAW)VGAP") {
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply polarisJawsApp/Db/delayed_jaws.template]: Are you sure it overwrites something? `TOP_LEVEL_JAW` is set to `POLJAWSET:JAWS5:` the `UNDERLYING_JAW` is `MOT:JAWS5`. Is it `POLJAWSET:JAWS5:` that is getting duplicated? If so I'm not sure where the other comes from? (I could be being dense here)

File: jawsManagerApp/Db/jaws_manager_single.template (hunk 1 above)
Comment (DominicOram) [reply jawsManagerApp/Db/jaws_manager_single.template]: They will always match the values in the underlying jaw, which is archived, the delayed version will not match the underlying one until the `SET` PV is called so it makes sense to archive both in that case.

**Pull request merged**